### PR TITLE
Add "type" and "accepts_webextensions" properties to browser data

### PR DESCRIPTION
**Pull request by queengooborg**
This PR moves some of the information that's defined in `test-browsers` to the browser schema, allowing us to define a browser's "type" (desktop, mobile, or server), and specify if web extensions are allowed in a similar fashion to flags.  Additionally, this PR updates the linter to utilize these parameters.

The idea is to make information we already use available to consumers, including MDN Web Docs.  The more information we can move out of our linter scripts, the better.

### CODE REVIEW

**Comment by lukewarlow** [comment]
This would be a massive change and probably one for a whole separate issue, but would it be worth thinking about changing from "browser" terminology to "runtime"? With nodejs, deno and potentially Cloudflare workers #14629 the browser term doesn't really fit anymore.

**Comment by queengooborg** [comment]
I've been thinking about this as well, since NodeJS and Deno aren't "browsers" but we've added them as "browsers".  Would you be up for opening a separate issue for that (or if there already is one, commenting on it)?

**Comment by foolip** [comment]
What do consumers say about the "type" addition? MDN already has this categorization, but would they be helped by putting it in the data as well? Would they really use it instead of just continuing to do what they currently do?

**Comment by queengooborg** [comment]
I think it would most certainly help them out, yes.  Looking through Yari's source code, I found the following bit of code:

https://github.com/mdn/yari/blob/ba8ed2f1403c332319e032eb1e9346b17668d068/client/src/document/ingredients/browser-compatibility-table/headers.tsx#L4-L17

This little bit of code looks pretty familiar though, doesn't it?  

https://github.com/mdn/browser-compat-data/blob/fc9df64939213a480a5d89324b1d0cba1d841f77/test/linter/test-browsers.js#L10-L24

These same lines of code in BCD are ones being removed and replaced in this PR, and there are many other lines in Yari we could simplify, which would allow for easier addition of browsers to Yari in the future.

File: test/linter/test-browsers-presence.js [hunk 1]
```diff
@@ -39,16 +25,46 @@ const browsers = {
...
+
+    let displayBrowsers = [
+      ...Object.keys(browsers).filter((b) =>
+        ['desktop', 'mobile'].includes(browsers[b].type),
+      ),
+      ,
+    ];
+    let requiredBrowsers = [
+      ...Object.keys(browsers).filter((b) => browsers[b].type == 'desktop'),
+    ];
```
Comment (caugner) [inline test/linter/test-browsers-presence.js:41]: What's the advantage of spreading the `Array.filter()` result into a fresh array?

```suggestion
    let displayBrowsers = Object.keys(browsers).filter((b) =>
        ['desktop', 'mobile'].includes(browsers[b].type));
    let requiredBrowsers = Object.keys(browsers).filter((b) => browsers[b].type == 'desktop');
```

**Review by caugner — CHANGES_REQUESTED** [review]

File: test/linter/test-browsers-presence.js [hunk 2]
```diff
@@ -39,16 +25,46 @@ const browsers = {
...
+      );
+    }
+
+    if (category === 'webextensions') {
+      displayBrowsers = displayBrowsers.filter(
+        (b) => browsers[b].accepts_webextensions,
+      );
+      requiredBrowsers = requiredBrowsers.filter(
+        (b) => browsers[b].accepts_webextensions,
+      );
```
[the remark this answers is not in the document]
Comment (queengooborg) [reply test/linter/test-browsers-presence.js:49]: Actually, this still only requires desktop browsers!  This is filtering that is applied atop of the the existing filtering, so `requiredBrowsers` filters for desktop browsers, then that array is filtered for browsers that accept webextensions.

**Pull request merged**